kuksa-sdk/src/test/kotlin/org/eclipse/kuksa/subscription/SubscriptionManagerTest.kt

```diff
@@ -0,0 +1,253 @@
+                val propertyObserverMock = mockk<PropertyObserver>(relaxed = true)
+                classUnderTest.subscribe(vssPath, fieldValue, propertyObserverMock)
+
+                and("When the FIELD_VALUE of Vehicle.Speed is updated") {
+                    databrokerApiInteraction.setRandomFloatValue(vssPath)
+
+                    then("The PropertyObserver is notified about the change") {
+                        verify(timeout = 100L) {
+                            propertyObserverMock.onPropertyChanged(vssPath, fieldValue, any())
+                        }
+                    }
+                }
+
+                `when`("Subscribing the same PropertyObserver to a different vssPath") {
+                    clearMocks(propertyObserverMock)
+
+                    val otherVssPath = "Vehicle.ADAS.CruiseControl.SpeedSet"
+                    classUnderTest.subscribe(otherVssPath, fieldValue, propertyObserverMock)
+
+                    and("Both values are updated") {
+                        databrokerApiInteraction.setRandomFloatValue(vssPath)
+                        databrokerApiInteraction.setRandomFloatValue(otherVssPath)
+
+                        then("The Observer is notified about both changes") {
+                            verify(timeout = 100L) {
+                                propertyObserverMock.onPropertyChanged(vssPath, fieldValue, any())
+                            }
+                            verify(timeout = 100L) {
+                                propertyObserverMock.onPropertyChanged(otherVssPath, fieldValue, any())
+                            }
+                        }
+                    }
+                }
+
+                `when`("Subscribing multiple (different) PropertyObserver to $vssPath") {
+                    clearMocks(propertyObserverMock)
+
+                    val propertyObserverMocks = mutableListOf<PropertyObserver>()
+                    repeat(10) {
+                        val otherPropertyObserverMock = mockk<PropertyObserver>(relaxed = true)
+                        propertyObserverMocks.add(otherPropertyObserverMock)
+
+                        classUnderTest.subscribe(vssPath, fieldValue, otherPropertyObserverMock)
+                    }
+
+                    and("When the FIELD_VALUE of Vehicle.Speed is updated") {
+                        val randomFloatValue = databrokerApiInteraction.setRandomFloatValue(vssPath)
+
+                        then("The PropertyObserver is only notified once") {
+                            propertyObserverMocks.forEach { propertyObserverMock ->
+                                val dataEntries = mutableListOf<DataEntry>()
+
+                                verify(timeout = 100L) {
+                                    propertyObserverMock.onPropertyChanged(vssPath, fieldValue, capture(dataEntries))
+                                }
+
+                                val count = dataEntries.count { it.value.float == randomFloatValue }
```

Comment:
Due to the asynchronous operation of #subscribe testing this stuff is a bit annoying.

When subscribing the observer it will be notified about the last known state, afterwards when the value changes it will be notified a second time, in the test these operations happen to shortly after each other, therefore it's hard to work with clearMocks or stuff like this

Maybe this is something which can be improved in the future